tests/integration/galley/statusupdate/statusupdate_test.go

```diff
@@ -0,0 +1,110 @@
+package statusupdate
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/label"
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("statusupdate_test", m).
+		Label(label.CustomSetup).
+		SetupOnEnv(environment.Kube, istio.Setup(nil, setupConfig)).
+		Run()
+}
+
+func setupConfig(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	cfg.Values["galley.enableAnalysis"] = "true"
+}
+
+const (
+	serviceRoleBindingYaml = `
+apiVersion: rbac.istio.io/v1alpha1
+kind: ServiceRoleBinding
+metadata:
+  name: service-role-binding
+  namespace: default
+spec:
+  mode: PERMISSIVE
+  roleRef:
+    kind: ServiceRole
+    name: service-role
+  subjects:
+  - user: '*'
+`
+	serviceRoleYaml = `
+apiVersion: rbac.istio.io/v1alpha1
+kind: ServiceRole
+metadata:
+  name: service-role
+  namespace: default
+spec:
+  rules:
+  - services: ["*"]
+`
+	namespace       = "default"
+	timeout         = "10s"
+	pollingInterval = "1s"
+)
+
+func TestStatusUpdate(t *testing.T) {
+	framework.NewTest(t).
+		// The status update implementation we're testing is kube-specific
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			g := NewGomegaWithT(t)
+
+			env := ctx.Environment().(*kube.Environment)
+
+			gvr := schema.GroupVersionResource{
+				Group:    "rbac.istio.io",
+				Version:  "v1alpha1",
+				Resource: "servicerolebindings",
+			}
+			name := "service-role-binding"
+
+			// Apply config that should generate a validation message
+			err := env.ApplyContents(namespace, serviceRoleBindingYaml)
+			defer func() { _ = env.DeleteContents(namespace, serviceRoleBindingYaml) }()
+
+			if err != nil {
+				t.Fatalf("Error applying serviceRoleBindingYaml for test scenario: %v", err)
+			}
+
+			// Verify we have the expected validation message
+			g.Eventually(func() string { return getStatus(t, env, gvr, namespace, name) }, timeout, pollingInterval).
```

Comment:
Minor nit: seems like you can move the inline `func()...` to a member and reuse it here and down in the second eventually call.

Reply:
Since the func needs to take no args I need it to have a closure, which is why I wasn't able to completely abstract it out. 

That said, I can create the func variable in-line once and reuse it, which is better.